requirements/dev.txt

```diff
@@ -8,7 +8,7 @@ pylint==1.7.2
 pytest==3.0.7
 py==1.4.33
 mock==2.0.0
-requests>=2.19.0
+requests==2.19.0
```

Comment:
No! why!

Reply:
These are test requirements and without pinning we fail some tests because the request version is in the user-agent headers: https://github.com/awslabs/aws-sam-cli/blob/develop/tests/functional/local/apigw/test_local_apigw_service.py#L629

```
"headers": {"Host": "0.0.0.0:33651", "User-Agent": "python-requests/2.19.0", ...
```

Requests library adds these headers. I had different versions of requests in my local Py3 vs Py2.

Reply:
I would rather handle this outside of this PR. Pip should not break for pinning (we already do this for pytest, py, mock, etc) in the dev.txt requirements. For developing with the cli, we recommend pyenv to allow multiple python versions for testing. Devs should not be installing this into an existing pyenv. I think the concern here is valid but very low for 'break if we pin'.

Reply:
Ahh my bad, I thought this was the base requirements.txt. Dev requirements are okay to be pinned